### DISCUSSION TITLE
AEBN, GayHotMovies and GayDVDEmpire Agents no longer work for me

### DISCUSSION THREAD

**Discussion by fourstix** [CodyBerenson/PGMA-Modernized#300] [Ideas]
The state legislature in North Carolina passed an age verification law for porn sites similar to laws in other states.  

Many porn sites now block Arkansas, Florida, Louisiana, Mississippi, Montana, North Carolina, Texas, Utah,  and Virginia. These three agent sites that I use now block all anonymous search requests that seem to originate from these locations and the PGMA agents no longer match for me.

But if I sign into these sites with a user id and password, I can make the exact same search request just fine.  It fails when the request is made by the agent.  (AEBN returns no results, GayDVDEmpore and GayHotMovies report an SSL error.)

From what I've read a VPN isn't a very good option for Plex.  Plex doesn't seem to work well with a VPN even with split-tunneling enabled. 

But I do have a user id and password for these three sites.  Would it be possible to upgrade the agents to have an optional a user/password field to sign into the site before making the search requests?

**Comment by CodyBerenson** [comment]
Hmmm....I wonder if this is related to why Google killed our Thumbor account for "violation" of their rules.....

**Comment by fourstix** [comment]
Could be.  It's hard to say, since I don't believe Google ever gives a specific reason for the violation.

**Comment by JPH71** [comment]
I will look into it...
I have a rather severe dislike of evangelicals... pesky little shits...

**Comment by CodyBerenson** [comment]
@fourstix If you're concerned about Plex and VPN, you only need to turn on your VPN when you're actually scraping.  I set mine to somewhere blue and free, or, overseas.

**Comment by fourstix** [comment]
@CodyBerenson  Thank you, But I'm not sure how to do that on my QNAP NAS box.  

I use PIA on my windows laptop and that's fairly simple to turn off and on, but the only option I see for QNAP is is based on OpenVPN, and it looks rather complicated to set up.

**Comment by fourstix** [comment]
@JPH71  Thank you, for looking at this.  Please let me know if there's anything I can do to help.

**Comment by CodyBerenson** [comment]
https://www.qnap.com/en/how-to/faq/article/how-to-set-up-a-vpn-server-on-qnap-nas-behind-the-router

https://www.youtube.com/watch?v=eZWI8Yc24JA

Looks relatively straight forward.  I did something similar for my TrueNAS server.  FWIW and IMO, its pretty dangerous not to be using VPN for pretty much everything.

**Comment by fourstix** [comment]
@CodyBerenson Ah, Great.  Thank you.  I'll check it out and see if I can set it up to work with PIA.

**Reply by fourstix** [reply]
@CodyBerenson  Using the information on the links you sent, I was able to set up QVPN to use PIA (Private Internet Access).  PIA has an OpenVPN configuration generator that made the set up relatively painless.  

Once I set it up to use a VPN server in California as the gateway server, the agents began to work again.  Plex seems to work okay.  

Thanks so much for your help.

**Reply by CodyBerenson** [reply]
That is GREAT!!!!  I will confess, setting up OpenVPN on TrueNAS took me bit, it was a little perplexing.  Really glad to hear it worked out.

**Comment by JPH71** [comment]
I think it would be a good idea - if you both wrote a statement of
procedure on how you did it and then Cody can upload them as help guides...

Cheers

Jason

**Reply by fourstix** [reply]
Hi @JPH71 and @CodyBerenson 
Good idea.  Here's the list of steps that I used to set up the VPN Client on QNAP for PIA.

1. Sign up for Private Internet Access (PIA) at https://www.privateinternetaccess.com

2. Download the OpenVPN configuration file from PIA.

    a) Login into the Private Internet Access website using your PIA username and password
    b) On your account page, click on Downloads in the left hand menu.
    c) Near the bottom of the Downloads page find the VPN Configuration settings section
    d) Under OpenVPN Configuration Generator click on the "Go To OpenVPN Generator" link
    e) On the next page choose OpenVPN 2.4 or newer, Linux operating system and then location
       port and use IP (address vs hostname) you wish to use.
    f) Click on the Generate button to download the config file.

3. Install QVPN from the Application Center

    a) Log in to QTS as an administrator.
    b) Open the App center.
    c) Install QVPN service.
    d) Start the QVPN service

4. Open QVPN and configure the VPN Client
    a) On the left hand menu click on VPN Connection Profiles, under VPN Client
    b) Click on the Add button and select OpenVPN from the dropdown menu.
    c) In the Create VPN Connection dialog, click on the browse button for Import .opvn file.
    d) Select the file you downloaded in step 2.
    e) Then click the Apply button
    f) In the pop up dialog box, enter your PIA username and password.
    g) Click on Apply to configure the VPN client.

5. Enable the PIA VPN client connection and set as default Gateway
    a) In the list of Other VPN Connection Profiles, locate the PIA VPN Connection profile you created in step 3.
    b) Click the button in the Action column to enable the PIA VPN Connection.
    c) Click on the Use VPN as NAS Default Gateway button at top of page.
    d) Click on the button labeled Enable VPN Gateway.
    e) Select the PIA VPN Connection in the VPN list.
    f) Click on the apply button.

6. Verify the VPN Connection 
   a) Click on the Overview menu item in the left hand menu
   b) On the Overview page, verify the VPN Connection you created is listed under Outgoing Connections
   c) In Plex, Refresh the Metadata for a Movie and verify the PGMA agents work as they should.

**Comment by CodyBerenson** [comment]
@fourstix your step-by-step has been included in the Usage Guide.  Thanks!

**Comment by JPH71** [comment]
Star!
